src/sentry/incidents/endpoints/serializers.py

```diff
@@ -151,189 +353,117 @@ def update(self, instance, validated_data):
-            elif target_type == AlertRuleTriggerAction.TargetType.USER:
-                try:
-                    user = User.objects.get(id=identifier)
-                except User.DoesNotExist:
-                    raise serializers.ValidationError("User does not exist")
-
-                if not OrganizationMember.objects.filter(
-                    organization=self.context["organization"], user=user
-                ).exists():
-                    raise serializers.ValidationError("User does not belong to this organization")
-        elif attrs.get("type") == AlertRuleTriggerAction.Type.SLACK:
-            if not attrs.get("integration"):
-                raise serializers.ValidationError(
-                    {"integration": "Integration must be provided for slack"}
-                )
-
-        return attrs
-
-    def create(self, validated_data):
-        return create_alert_rule_trigger_action(trigger=self.context["trigger"], **validated_data)
+            # TODO: Remove this, just temporary while we're supporting both fields.
+            if "aggregation" not in validated_data:
+                raise serializers.ValidationError("aggregation is required")
 
-    def _remove_unchanged_fields(self, instance, validated_data):
-        changed = False
-        if (
-            validated_data.get("type", instance.type) == AlertRuleTriggerAction.Type.SLACK.value
-            and validated_data["target_identifier"] != instance.target_display
-        ):
-            changed = True
-        for field_name, value in list(six.iteritems(validated_data)):
-            # Remove any fields that haven't actually changed
-            if isinstance(value, Enum):
-                value = value.value
-            if getattr(instance, field_name) != value:
-                changed = True
-                break
-        return validated_data if changed else {}
+            triggers_data = validated_data.pop('triggers')
+            # TODO: User super.create and don't duplicate the aggreagation + duplicate name check?
+            alert_rule = create_alert_rule(organization=self.context["organization"], **validated_data)
+            for trigger_data in triggers_data:
+                trigger_actions_data = trigger_data.pop("actions")
+                trigger = create_alert_rule_trigger(alert_rule=alert_rule, **trigger_data)
+                for actions_data in trigger_actions_data:
+                    create_alert_rule_trigger_action(trigger=trigger, **actions_data)
+            return alert_rule
+        except AlertRuleNameAlreadyUsedError:
+            raise serializers.ValidationError("This name is already in use for this project")
 
     def update(self, instance, validated_data):
         validated_data = self._remove_unchanged_fields(instance, validated_data)
-        return update_alert_rule_trigger_action(instance, **validated_data)
+        triggers_data = validated_data.pop("triggers")
```

Comment:
Will we handle triggers/actions being removed here?

Reply:
Hmm...that's a good question. Or should we leave the individual delete for triggers/actions? 

Otherwise we'd need to do some magic on fetching the triggers/actions that weren't just updated/created or that exist and aren't in the list, etc..

Reply:
It's being handled now! Added a couple of tests for this as well.